### DISCUSSION TITLE
Does assume_emoji_presentation allow support of COLR/CPAL color fonts?

### DISCUSSION THREAD

**Discussion by UliTroyo** [wezterm/wezterm#2735] [Q&A] [answered]
Hi! I was messing around with fonts and font settings, and I was not able to load any color fonts, but I can't figure out if it's because they're just not supported or because I'm doing something wrong. I want to be able to take a font like, for example, [Quantum Ukraine](https://www.fontspace.com/quantum-ukraine-font-f80782) and display it. (This isn't a good font or anything, it just has color, and I thought I could load it as an emoji font. I'm making my own font, and it uses OpenType color tables, and I wanted to know if I would be able to use it in WezTerm.) Is there something I could try? What aspects of a font does 'assume_emoji_presentation' look at? Thanks for your time regardless :)

**Comment by wez** [comment, marked as answer]
When a font has a colored face wezterm assumes that the font has emoji presentation.

Each cluster of text that is to be shaped is tagged with its presentation, which can be one of:

  * Explicit emoji presentation (eg: text that has an explicit Emoji presentation variation selector)
  * Explicit text presentation (eg: text that has an explicit Text presentation variation selector)
  * No explicit presentation; presentation is inferred from the unicode codepoint properties

When shaping, up to two passes are made over the list of fonts in the fallback:

 * First pass is to find a font whose emoji presentation is equal to the presentation of the text that has glyphs for the text.  The font fallback list is enumerated until a match is found. If a match is found, shaping proceeds with that match
 * If no match was found, a second pass is made where the presentation of the font is ignored, allowing any of the fonts that have appropriate glyphs to be used. 

This two-pass lookup is effective for resolving emoji (which are relatively uncommon) from colored and monochrome emoji fonts.

It is imperfect for the font you want to try because the font is assumed to have emoji presentation, which causes a shaping of text to skip it and find the next font in the fallback and use that instead.

You can override whether a font should be consider to have emoji presentation:

```lua
  font = wezterm.font{family="Quantum Ukraine", assume_emoji_presentation=false},
```

and it shows up colored correctly in wezterm.

**Reply by UliTroyo** [reply]
Thanks! Yeah, this absolutely solved it. Thank you so much for the explanation!
